pkg/provider/github/github.go

```diff
@@ -238,12 +239,37 @@ func (w *GitHub) HandleHook(ctx context.Context, req *http.Request) (int, error)
 func (w *GitHub) handleEvent(ctx context.Context, client *github.Client, event interface{}, receiver *webhookv1.GitWatcher) (int, error) {
 	execution := initExecution(receiver)
 	switch event.(type) {
+	case *github.CreateEvent:
+		if receiver.Spec.Tag == false {
+			return http.StatusUnprocessableEntity, fmt.Errorf("create events only support tag")
+		}
+		parsed := event.(*github.CreateEvent)
+		if parsed.Ref == nil {
+			return http.StatusUnprocessableEntity, errors.New("create event has invalid tag ref")
+		}
+		if parsed.GetRefType() != "tag" {
+			return http.StatusUnprocessableEntity, errors.New("create event has non-tag type")
+		}
+		execution.Spec.Tag = *parsed.Ref
+		err := git.TagMatch(receiver.Spec.TagQuery, receiver.Spec.TagExclude, execution.Spec.Tag)
+		if err != nil {
+			return http.StatusUnprocessableEntity, err
+		}
+		if parsed.Sender != nil {
+			execution.Spec.Author = safeString(parsed.Sender.Login)
+			execution.Spec.AuthorEmail = safeString(parsed.Sender.Email)
+			execution.Spec.AuthorAvatar = safeString(parsed.Sender.AvatarURL)
+		}
+
 	case *github.PushEvent:
```

Comment:
Does push event still receive tags? I thought we decided that push only watches commits while create watches tags.

Reply:
Push includes new tags yes, even via tag creation in UI. Create watches both tags and new branches so I have to exclude branches.

Reply:
fixed